src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/producttypes/ProductTypeSyncIT.java

```diff
@@ -0,0 +1,145 @@
+            .stream()
+            .map(ProductTypeDraftBuilder::of)
+            .map(ProductTypeDraftBuilder::build)
+            .collect(Collectors.toList());
+
+        final List<String> errorMessages = new ArrayList<>();
+        final List<Throwable> exceptions = new ArrayList<>();
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .errorCallback((error, throwable) -> {
+                errorMessages.add(error);
+                exceptions.add(throwable);
+            })
+            .build();
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(productTypeSyncOptions);
+
+        // test
+        final ProductTypeSyncStatistics productTypeSyncStatistics = productTypeSync
+            .sync(productTypeDrafts)
+            .toCompletableFuture().join();
+
+        // assertion
+        assertThat(errorMessages).isEmpty();
+        assertThat(exceptions).isEmpty();
+        assertThat(productTypeSyncStatistics).hasValues(2, 1, 0, 0);
+        assertThat(productTypeSyncStatistics
+            .getReportMessage())
+            .isEqualTo("Summary: 2 product types were processed in total"
+                + " (1 created, 0 updated and 0 failed to sync).");
+    }
+
+    @Test
+    public void sync_WithUpdates_ShouldReturnProperStatistics() {
+        // preparation
+        final List<ProductType> productTypes = CTP_SOURCE_CLIENT
+            .execute(ProductTypeQuery.of())
+            .toCompletableFuture().join().getResults();
+
+        final List<ProductTypeDraft> productTypeDrafts = productTypes
+            .stream()
+            .map(productType -> {
+                final List<AttributeDefinitionDraft> attributeDefinitionDrafts = productType
+                    .getAttributes()
+                    .stream()
+                    .map(attribute -> AttributeDefinitionDraftBuilder.of(attribute).build())
+                    .collect(Collectors.toList());
+
+                return ProductTypeDraftBuilder
+                    .of(
+                        productType.getKey(),
+                        productType.getName() + "_updated", // name updated
+                        productType.getDescription(),
+                        attributeDefinitionDrafts)
+                    .build();
+            })
+            .collect(Collectors.toList());
+
+        final List<String> errorMessages = new ArrayList<>();
+        final List<Throwable> exceptions = new ArrayList<>();
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .errorCallback((error, throwable) -> {
+                errorMessages.add(error);
+                exceptions.add(throwable);
+            })
+            .build();
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(productTypeSyncOptions);
```

Comment:
Maybe you can move this to the setup method, to make the code more readable

Reply:
Actually, I'd prefer to leave it like this, since it could be that there is a new coming test which have different options for the sync. I think it's better if every test is isolated and has it's own state.